### PR TITLE
Implement cosmic_text as the text renderer and editor

### LINKED ISSUE #95: Investigate using Cosmic Text
In order to make yakui's text rendering better, we should consider using Cosmic Text as the default text backend!

I don't know if it's well-suited for games, but it sure looks pretty.

### PR DESCRIPTION
**Pull request by msparkles**
Closes #95

### CODE REVIEW

**Comment by msparkles** [comment]
image

Weird text bug on yakui demo

**Comment by msparkles** [comment]
Unsure of why the tests are failing!

**Comment by msparkles** [comment]
Need input on the tests and might need help on the Vulkan code (we modified the text shader on wgpu to support coloured glyphs).

Otherwise should be ready for review!

File: crates/yakui-widgets/src/widgets/textbox.rs [hunk 1]
```diff
@@ -128,7 +127,7 @@ pub struct TextBoxWidget {
 pub struct TextBoxResponse {
     pub render_text: RenderText,
     pub scroll: Option<cosmic_text::Scroll>,
-    pub text: Option<String>,
+    pub text: String,
```
Comment (LPGhatguy) [inline crates/yakui-widgets/src/widgets/textbox.rs:130]: I don't think we can make a change like this: `Option<String>` is important to know whether the text changed or not.

Comment (msparkles) [reply crates/yakui-widgets/src/widgets/textbox.rs:130]: Hmmmmmmmmmmmmmmmmmmmmmmm

If that's the reason for an Option<String>, we'd need to somehow know when the text has been modified.

**Comment by msparkles** [comment]
Hm! Either I'm doing something wrong here, or whatever #125 aims to fix is hitting me hard here

**Comment by LPGhatguy** [comment]
Even after #125 or a similar fix, you'll still run into that until we fix widget keys. More details in Discord!

Ideally, widgets are able to react to props changing so that it isn't a major issue to reuse widget state but sometimes that isn't possible.

**Comment by msparkles** [comment]
we've made a branch that contains code to solve the issue. but it depends on this pr because we were too lazy to find a reprod on main that actually does have the issue, lol!

**Comment by msparkles** [comment]
AAAaahhhhh. hi-dpi actually works now. This has become like 40 commits, god damn it

**Comment by msparkles** [comment]
oops squashed them wrongly i think

**Comment by msparkles** [comment]
apparently the text fixes broke the height of texts on certain fonts, so the actual thing we should've done was to fix the tests :100:

**Comment by msparkles** [comment]
Closed as per #194

**Pull request closed without merging**